src/main/java/subway/LineService.java

```diff
@@ -0,0 +1,53 @@
+package subway;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Service
+@Transactional(readOnly = true)
+public class LineService {
+
+    private final LineRepository lineRepository;
+
+    public LineService(LineRepository lineRepository) {
+        this.lineRepository = lineRepository;
+    }
+
+    public Line createLine(LineCreateRequest request) {
+        Line line = Line.builder()
+                .name(request.getName())
+                .color(request.getColor())
+                .upStationId(request.getUpStationId())
+                .downStationId(request.getDownStationId())
+                .distance(request.getDistance())
+                .build();
+        return lineRepository.save(line);
+
+    }
+
+
+    public Line readLine(Long id) throws HttpException {
+        return lineRepository.findById(id).orElseThrow(() -> new HttpException(ErrorCode.BAD_REQUEST));
+    }
+
+
+    public List<Line> readLines() {
+        return lineRepository.findAll();
+    }
+
+    @Transactional
+    public void updateLine(LineUpdateDTO dto) throws HttpException {
+        Line line = lineRepository.findById(dto.getId())
+                .orElseThrow(() -> new HttpException(ErrorCode.BAD_REQUEST));
+        line.changeColor(dto.getColor());
+        line.changeName(dto.getName());
+        lineRepository.save(line);
+    }
+
+    @Transactional
+    public void deleteLine(Long id) throws HttpException {
```

Comment:
실제로 예외가 던져지지 않는다면 제거해도 되지 않을까요?
```suggestion
    public void deleteLine(Long id) {
```